mod/primitives/pkg/encoding/hex/bytes.go

```diff
@@ -28,31 +28,36 @@ import (
 
 var ErrInvalidHexStringLength = errors.New("invalid hex string length")
 
-func EncodeBytes[B ~[]byte](b B) []byte {
-	result := make([]byte, len(b)*2+prefixLen)
-	copy(result, prefix)
-	hex.Encode(result[prefixLen:], b)
-	return result
+func FromBytes[B ~[]byte](b B) string {
+	return prefix + hex.EncodeToString(b)
 }
 
-func UnmarshalByteText(input []byte) ([]byte, error) {
```

Comment:
replaced by `ToBytes` since:

1. `formatAndValidateText` is equivalent to `IsValidHex` since:
  - inputs are never zero-lenght
  - The check on lenght being even is implicitly done by `DecodeString`
2. `hex.Decode` behaviour with  `dec := make([]byte, len(raw)/encDecRatio)` matches `hex.DecodeString(`